gems/aws-sdk-core/lib/aws-sdk-core/shared_config_file_standardizer.rb

```diff
@@ -0,0 +1,155 @@
+# frozen_string_literal: true
+
+module AWS::SDK::Core
+  # Converts a raw map of maps into a map of profiles.
+  # @api private
+  class SharedConfigFileStandardizer
+    def initialize(raw_file, file_type)
+      @raw_file = raw_file
+      @file_type = file_type
+    end
+
+    def standardize
+      standardize_profile_file
+      if @file_type == :config
+        [@standardized_profiles, @standardized_sso_sessions]
+      else
+        @standardized_profiles
+      end
+    end
+
+    private
+
+    # rubocop:disable Metrics
+    def standardize_profile_file
```

Comment:
naming nit, could this just be standarize_file? (would have said stadardize_shared_config_file but thats rather long).